### PR TITLE
make structuredDiffForFile for BitBucketServer the same as the others

### PR DESCRIPTION
**Pull request by berlysia**
#1200 

1. Adjust `structuredDiffForFile` for BitBucketServer the same as the others
2. Remove `any` from `StructuredDiff`

### CODE REVIEW

File: source/platforms/bitbucket_server/BitBucketServerGit.ts [hunk 1]
```diff
@@ -131,23 +131,56 @@ const bitBucketServerChangesToGitJSONDSL = (
...
   // We need all changed lines with it's type. It will convert hunk segment lines to flatten changed lines.
-  const segmentValues = { ADDED: "add", CONTEXT: "normal", REMOVED: "del" }
+  const segmentValues = { ADDED: "add", CONTEXT: "normal", REMOVED: "del" } as const
   return diffs.map(diff => ({
     from: diff.source && diff.source.toString,
     to: diff.destination && diff.destination.toString,
     chunks:
       diff.hunks &&
       diff.hunks.map(hunk => ({
+        content: `@@ -${hunk.sourceLine},${hunk.sourceSpan} +${hunk.destinationLine},${hunk.destinationSpan} @@`,
```
Comment (berlysia) [inline source/platforms/bitbucket_server/BitBucketServerGit.ts:141]: 💭 Is that too much? https://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html

File: source/platforms/bitbucket_server/BitBucketServerGit.ts [hunk 2]
```diff
@@ -131,23 +131,56 @@ const bitBucketServerChangesToGitJSONDSL = (
...
+            if (type === "normal") {
+              return segment.lines.map(line => {
+                return {
+                  type,
+                  normal: true as const,
+                  content: line.line,
+                  ln1: line.source,
+                  ln2: line.destination,
+                }
+              })
```
Comment (berlysia) [inline source/platforms/bitbucket_server/BitBucketServerGit.ts:178]: These transformation is based on https://docs.atlassian.com/bitbucket-server/rest/7.19.3/bitbucket-rest.html#idp261

**Comment by berlysia** [comment]
`yarn declaration` is now passed in my local env

**Comment by orta** [comment]
Perfect, all this looks great - thanks!

File: source/platforms/bitbucket_server/BitBucketServerGit.ts (hunk 1 above)
Comment (orta) [reply source/platforms/bitbucket_server/BitBucketServerGit.ts:141]: Nah, that looks right enough to me

**Pull request merged**